crates/rytest/src/phases/execution.rs

```diff
@@ -1,12 +1,102 @@
 use anyhow::Result;
 use pyo3::prelude::*;
-use pyo3::types::{PyIterator, PyMapping, PyString};
+use pyo3::types::{PyIterator, PyMapping};
 use pyo3::{indoc::indoc, types::PyTuple};
+use std::path::PathBuf;
 use std::{env, fs, path::Path, sync::mpsc};
 
 use crate::python;
 use crate::TestCase;
 
+fn file_contains_fixture(file_path: &Path, fixture_name: &str) -> bool {
+    if let Ok(content) = fs::read_to_string(file_path) {
+        let def_pattern = format!("def {}(", fixture_name);
+        let fixture_pattern = format!("fixture(name=\"{}\")", fixture_name);
+        return content.contains(&def_pattern) || content.contains(&fixture_pattern);
```

Comment:
Would this capture any function that matches the name?  

Should it instead validate either of the two patterns, and if it's the first, validate it has a fixture decorator?

Also do we think this might benefit from the AST pattern we use elsewhere?